components/calendar/calendar.js

```diff
@@ -0,0 +1,480 @@
+		}
+	}
+	return numDaysFromLastMonthToShowThisMonth;
+}
+
+export function getNumberOfDaysInMonth(month, year) {
+	return new Date(year, month + 1, 0).getDate();
+}
+
+export function getNumberOfDaysToSameWeekPrevMonth(month, year) {
+	const prevMonth = getPrevMonth(month);
+	const prevMonthYear = prevMonth === 11 ? year - 1 : year;
+	const numDaysPrevMonth = getNumberOfDaysInMonth(prevMonth, prevMonthYear);
+	const numDaysFromPrevPrevMonthToShowPrevMonth = getNumberOfDaysFromPrevMonthToShow(prevMonth, prevMonthYear);
+	const numWeeksPrevMonth = Math.ceil((numDaysFromPrevPrevMonthToShowPrevMonth + numDaysPrevMonth) / 7);
+
+	const firstDayOfMonth = new Date(year, month, 1).getDay();
+	const hasDaysFromPrevMonth = (firstDayOfMonth !== firstDayOfWeek);
+	return daysInWeek * (numWeeksPrevMonth - ((hasDaysFromPrevMonth) ? 1 : 0));
+}
+
+export function getPrevMonth(month) {
+	return (month === 0) ? 11 : (month - 1);
+}
+
+export function parseDate(val) {
+	if (!val) return null;
+
+	let date = val;
+	if (val.includes('T')) {
+		const dateTimeParts = val.split('T');
+		date = dateTimeParts[0];
+	}
+	const dateParts = date.split('-');
+	if (dateParts.length !== 3) {
+		throw new Error('Invalid selected-value date input: Expected format is YYYY-MM-DDTHH:mm:ss.sssZ or YYYY-MM-DD');
+	}
+
+	return new Date(dateParts[0], parseInt(dateParts[1]) - 1, dateParts[2]);
+}
+
+class Calendar extends LocalizeStaticMixin(LitElement) {
+
+	static get properties() {
+		return {
+			selectedValue: { type: String, attribute: 'selected-value' },
+			summary: { type: String },
+			_focusDate: { type: Object },
+			_shownMonth: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return [calendarStyles, bodySmallStyles, heading4Styles];
+	}
+
+	static get resources() {
+		return {
+			'ar': { show: 'Show {month} ' },
+			'en': { show: 'Show {month}' },
+			'es': { show: 'Show {month} ' },
+			'fr': { show: 'Show {month} ' },
+			'ja': { show: 'Show {month} ' },
+			'ko': { show: 'Show {month} ' },
+			'nl': { show: 'Show {month} ' },
+			'pt': { show: 'Show {month} ' },
+			'sv': { show: 'Show {month} ' },
+			'tr': { show: 'Show {month} ' },
+			'zh': { show: 'Show {month} ' },
+			'zh-tw': { show: 'Show {month} ' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		this._labelId = getUniqueId();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this._today = new Date();
+		if (this.selectedValue) {
+			const selectedValueDate = parseDate(this.selectedValue);
+			this._focusDate = new Date(selectedValueDate.getFullYear(), selectedValueDate.getMonth(), selectedValueDate.getDate());
+			this._shownMonth = selectedValueDate.getMonth();
+			this._shownYear = selectedValueDate.getFullYear();
+		} else {
+			this._focusDate = new Date(this._today.getFullYear(), this._today.getMonth(), 1);
+			this._shownMonth = this._today.getMonth();
+			this._shownYear = this._today.getFullYear();
+		}
```

Comment:
Nit: this could be re-written as:
```
const date = this.selectedValue ? parseDate(this.selectedValue) : this._today;
this._focusDate = new Date(date.getFullYear(), date.getMonth(), 1);
this._shownMonth = date.getMonth();
this._shownYear = date.getFullYear();
```